src/EFCore.PG/Query/ExpressionVisitors/NpgsqlSqlTranslatingExpressionVisitor.cs

```diff
@@ -122,101 +81,80 @@ protected override Expression VisitBinary(BinaryExpression expression)
         /// An '= ANY' expression or null.
         /// </returns>
         [CanBeNull]
-        protected virtual Expression VisitEqualsAny([NotNull] SubQueryExpression expression)
-        {
-            var subQueryModel = expression.QueryModel;
-            var fromExpression = subQueryModel.MainFromClause.FromExpression;
+        protected override Expression VisitSubQuery(SubQueryExpression expression)
+            => base.VisitSubQuery(expression) ?? VisitArraySubQuery(expression);
 
-            var properties = MemberAccessBindingExpressionVisitor.GetPropertyPath(
-                fromExpression, _queryModelVisitor.QueryCompilationContext, out _);
+        /// <inheritdoc />
+        protected override Expression VisitUnary(UnaryExpression expression)
+            => Visit(expression.Operand) is Expression operand
```

Comment:
This seems wrong, isn't `UnaryExpression.Operand` always an expression? That means you're always translating all unary expressions....

Reply:
`UnaryExpression.Operand` is indeed always an expression, but the result of `Visit(UnaryExpression.Operand)` can be `null` (which `VisitUnary` is not supposed to return). 

Since we need to check the result for `null` before reconstructing/skipping, I would prefer to use pattern matching here. Since pattern matching only makes the variable available in the `true` branch, there's no chance for a `null` result to accidentally be passed back in the `false` branch.

> always translating all unary expressions....

Unless it's an expression we care about (and have written a translator for), I believe that calling `Visit` should either:
1. return `null`, in which case we just move onto the base method; or,
2. return what the base method would have otherwise returned.

To be on the safe-side, I'll re-verify this works as described after the next commit.